libgalois/include/galois/analytics/connected_components/connected_components.h

```diff
@@ -0,0 +1,156 @@
+#ifndef GALOIS_LIBGALOIS_GALOIS_ANALYTICS_CONNECTEDCOMPONENTS_CONNECTEDCOMPONENTS_H_
+#define GALOIS_LIBGALOIS_GALOIS_ANALYTICS_CONNECTEDCOMPONENTS_CONNECTEDCOMPONENTS_H_
+
+#include <iostream>
+
+#include <galois/analytics/Plan.h>
+
+#include "galois/AtomicHelpers.h"
+#include "galois/analytics/Utils.h"
+
+// API
+
+namespace galois::analytics {
+
+/// A computational plan to for ConnectedComponents, specifying the algorithm and any
+/// parameters associated with it.
+class ConnectedComponentsPlan : Plan {
+public:
+  /// Algorithm selectors for Connected-components
+  enum Algorithm {
+    kSerial,
+    kLabelProp,
+    kSynchronous,
+    kAsynchronous,
+    kEdgeAsynchronous,
+    kEdgeTiledAsynchronous,
+    kBlockedAsynchronous,
+    kAfforest,
+    kEdgeAfforest,
+    kEdgeTiledAfforest,
+    kAutomatic,
```

Comment:
Automatic appears to be a duplicate of kAfforest. Let's just remove it everywhere.